diff --git a/ngx_http_upstream_check_module.c b/ngx_http_upstream_check_module.c
index 1a7cde7..f99982f 100644
--- a/ngx_http_upstream_check_module.c
+++ b/ngx_http_upstream_check_module.c
@@ -4062,6 +4062,10 @@ ngx_http_upstream_check_init_process(ngx_cycle_t *cycle)
 {
     ngx_http_upstream_check_main_conf_t *ucmcf;
 
+    if (ngx_process != NGX_PROCESS_WORKER) {
+        return NGX_OK;
+    }
+
     ucmcf = ngx_http_cycle_get_module_main_conf(cycle, ngx_http_upstream_check_module);
     if (ucmcf == NULL) {
         return NGX_OK;
